cmd/translatesfx/translatesfx/otel.go

```diff
@@ -60,64 +60,81 @@ func newOtelCfg() *otelCfg {
 }
 
 func translateFilters(sa saCfgInfo, otel *otelCfg) {
-	if sa.metricsToExclude == nil {
+	if sa.metricsToExclude == nil && sa.metricsToInclude == nil {
 		return
 	}
+
 	otel.Processors[filterProc] = map[string]interface{}{
 		"metrics": map[string]interface{}{
 			"exclude": map[string]interface{}{
 				"match_type":  "expr",
-				"expressions": saFiltersToExpr(sa.metricsToExclude),
+				"expressions": saExcludesToExpr(sa.metricsToExclude, sa.metricsToInclude),
 			},
 		},
 	}
 	otel.Service.Pipelines["metrics"].appendProcessor(filterProc)
 }
 
-func saFiltersToExpr(excludes []interface{}) []string {
+func saExcludesToExpr(excludes []interface{}, includes []interface{}) []string {
+	includesExpr := saIncludesToExpr(includes)
 	var out []string
-	for _, excludeV := range excludes {
-		exclude, ok := excludeV.(map[interface{}]interface{})
-		if !ok {
-			return nil
+	for _, v := range excludes {
+		line := filterToExpr(v.(map[interface{}]interface{}), false)
+		if includesExpr != "" {
+			line += " and (" + includesExpr + ")"
 		}
-		var names []interface{}
-		namesV, ok := exclude["metricNames"]
-		if !ok {
-			if nameV, metricNameOK := exclude["metricName"]; metricNameOK {
-				names = []interface{}{nameV}
-			} else {
-				return nil
-			}
-		} else {
-			names = namesV.([]interface{})
+		out = append(out, line)
+	}
+	return out
+}
+
+func saIncludesToExpr(includes []interface{}) string {
+	out := ""
+	for _, includeV := range includes {
+		line := filterToExpr(includeV.(map[interface{}]interface{}), true)
+		if out != "" {
+			out += " and "
 		}
-		line := metricNamesToExpr(names)
+		out += line
+	}
+	return out
+}
 
-		dimsV, ok := exclude["dimensions"]
-		var dims map[interface{}]interface{}
-		if ok && dimsV != nil {
-			dims = dimsV.(map[interface{}]interface{})
+func filterToExpr(pattern map[interface{}]interface{}, flipNegation bool) string {
+	var names []interface{}
+	namesV, ok := pattern["metricNames"]
+	if !ok {
+		if nameV, metricNameOK := pattern["metricName"]; metricNameOK {
+			names = []interface{}{nameV}
 		}
-		dimExpr := dimsToExpr(dims)
-		if dimExpr != "" {
-			if line != "" {
-				line += " and "
-			}
-			line += "(" + dimExpr + ")"
+	} else {
+		names = namesV.([]interface{})
```

Comment:
I think I've commented on something like this before, but would it be worthwhile to validate the SA config on the way in w/ `yaml.UnmarshalStrict()` and https://github.com/signalfx/signalfx-agent/blob/main/pkg/core/config/validation/validator.go#L33
 so that untested type assertions are less prone to hit conversion issues?

Reply:
Yeah, I've been thinking that may make things cleaner. I'll try it.